src/pages/Dashboard/widgets/BatchTransactions/BatchTransactions.tsx

```diff
@@ -32,109 +38,91 @@ export const BatchTransactions = () => {
-    transactionsOrder
+    signedSessionId: currentSessionId
   });
 
   // this process will not go through useSendSignedTransactions
   // it will automatically sign and send transactions
-  const signAndAutoSendBatchTransactions = async () => {
+  const { signAndAutoSendBatchTransactions } =
+    useSignAndAutoSendBatchTransactions();
+
+  const { sendBatchTransactions } = useSendBatchTransactions();
+  const { swapAndLockTokens } = useSwapAndLockTokens();
+
+  // If manual batch transactions are executed, track the batchId
+  useEffect(() => {
+    if (batchId) {
+      setTrackBatchId(batchId);
+    }
+  }, [batchId]);
+
+  useEffect(() => {
+    if (trackBatchId && batches[trackBatchId]) {
+      setStateTransactions(batches[trackBatchId].transactions.flat());
+    }
+  }, [trackBatchId, batches]);
+
+  const executeSignAndAutoSendBatchTransactions = async () => {
     setSendBatchTransactionsOnDemand(false);
 
-    const payload = getBatchTransactions(address);
-    const { transactions } = payload;
-
-    const groupedTransactions = [
-      [transactions[0]],
-      [transactions[1], transactions[2]],
-      [transactions[3], transactions[4]]
-    ];
-
-    const { batchId: currentBatchId, error } = await sendBatchTransactions({
-      transactions: groupedTransactions,
-      callbackRoute: window.location.pathname,
-      customTransactionInformation: { redirectAfterSign: true },
-      transactionsDisplayInfo: {
-        processingMessage: 'Processing transactions',
-        errorMessage: 'An error has occurred during transaction execution',
-        successMessage: 'Batch transactions successful'
-      }
+    const { batchId } = await signAndAutoSendBatchTransactions({
+      address,
+      nonce: account.nonce,
+      chainID: network.chainID,
+      callbackRoute
     });
 
-    if (error) {
-      console.error('Could not execute transactions', error);
+    if (!batchId) {
       return;
     }
 
-    sessionStorage.setItem(SessionEnum.batchId, currentBatchId);
-    setTrackBatchId(currentBatchId);
-  };
-
-  const executeBatchTransactions = () => {
-    const payload = getBatchTransactions(address);
-    executeTransactions(payload);
-  };
-
-  const executeSwapAndLockTransactions = () => {
-    const payload = getSwapAndLockTransactions(address);
-    executeTransactions(payload);
+    setTrackBatchId(batchId);
   };
 
-  const executeTransactions = async (payload: BatchTransactionsType) => {
+  const executeBatchTransactions = async () => {
     setSendBatchTransactionsOnDemand(true);
-    const { transactions, order } = payload;
-
-    const { sessionId, error } = await sendTransactions({
-      transactions,
-      signWithoutSending: true,
-      callbackRoute: window.location.pathname,
-      customTransactionInformation: { redirectAfterSign: true }
+    const { newBatchSessionId, sessionId } = await sendBatchTransactions({
+      address,
+      nonce: account.nonce,
+      chainID: network.chainID,
```

Comment:
aici de regula foloseam `chainId` de verificat

Reply:
it would be strange to use different usage of chainID / chainId. The interface is using chainID. Using a different name it will create confusion and might not be clear enough when using it